session_4/blogspot/app/views/articles/show.html.erb

```diff
@@ -1,10 +1,10 @@
-<div class="jumbotron">
-  <% if @article.image.attached? %>
-    <div class="text-center">
-      <%= image_tag(@article.image, class: "img-fluid shadow-lg mb-5 rounded", style: "max-width:100%;", alt: "No Image Found") %>
-    </div>
-  <% end %>
+<% if @article.image.attached? %>
```

Comment:
The "blurring" was because everything was inside the jumbotron div. I haven't tested this but it should work. Fix any errors you face and replace the contents of your file with:

```erb
<div class="jumbotron mb-2">
  <% if @article.image.attached? %>
    <div class="text-center">
      <%= image_tag(@article.image, class: "img-fluid shadow-lg mb-5 rounded", style: "max-width:100%;", alt: "No Image Found") %>
    </div>
  <% end %>

  <h1 class="display-4">
    <%= @article.title %>
  </h1>
</div>

<p class="lead">
  <strong>Topic:</strong>
  <%= @article.topic %>
</p>

<p class="lead">
  <strong>Tags:</strong>
  <%= @article.tags %>
</p>

<hr class="my-4">

<p class="lead">
  <%= @article.content %>
</p>

<% unless request.format.to_sym == :pdf %>
  <% if can? :edit, @article %>
    <%= link_to "DOWNLOAD ARTICLE", article_path(@article, format: :pdf), class:"btn btn-outline-dark", target: "_blank" %> | 
    <%= link_to 'Edit', edit_article_path(@article), class:"btn btn-outline-secondary" %> |
    <%= link_to 'Delete', @article, method: :delete, data: { confirm: 'Are you sure?' }, class:"btn btn-outline-danger" %> | 
  <% end %>
  <%= link_to 'Back', articles_path, class:"btn btn-outline-info" %>
<% end %>
```

Reply:
Yeah..It works fine even when I include the complete article details except the buttons. but i didn't understood why it happened and how it's resolved now. Like moving some buttons aur paragraph out of jumbotron resolved it?

Reply:
Ohh..It's not working.. Actually what i did was to compare between the code and their result in pdf. I pasted them one after other. So Your code was after the previous code. So when I generated the pdf On first page it was blurred but on the second page(your code), it was clear. Now, when i tried puting my code below and yours above. It's showing clear for mine also. Also, when i use any of the code single, it shows blurred only

Reply:
In summary, it's working when we somehow reach second page of the pdf, but not on the first pdf

Reply:
@abhishekkumar2718